### PR TITLE
(#358) implement console shortcuts for copy, cut, paste, clear

### LINKED ISSUE #358: Console does not support copy, cut, paste operations */
blocked by #350

### PR DESCRIPTION
**Pull request by isidentical**
Resolves #358

### CODE REVIEW

File: src/ui/edit_field.c [hunk 1]
```diff
@@ -449,6 +449,13 @@ void edit_field_replace(Edit_field *edit_field, const char *text)
     }
 }
 
+void edit_field_append(Edit_field *edit_field, const char *text)
+{
+    size_t n = strlen(text);
+    edit_field->buffer = strcat(edit_field->buffer, text);
```
Comment (rexim) [inline src/ui/edit_field.c]: Please don't use `strcat` it's prone to buffer overflows. Please use something like `strncat`.

File: src/ui/console_log.c [hunk 2]
```diff
@@ -107,3 +107,14 @@ int console_log_push_line(Console_Log *console_log,
...
     return 0;
 }
+
+void console_log_clear(Console_Log *console_log)
+{
+    trace_assert(console_log);
+    console_log->cursor = 0;
+    for (size_t i = 0; i < console_log->capacity; ++i) {
+        if (console_log->buffer[i]) {
+            console_log->buffer[i] = 0;
```
Comment (rexim) [inline src/ui/console_log.c]: Sorry, there is no GC in C. You can't just NULL out the buffers like that without leaking the memory. Free the buffer before NULL-ing it. See `destroy_console_log`.

File: src/ui/console.c [hunk 3]
```diff
@@ -207,6 +206,31 @@ int console_handle_event(Console *console,
...
+            }
+        } break;
+
+       case SDLK_y: {
+            if (event->key.keysym.mod & KMOD_CTRL) {
+                char *text = SDL_GetClipboardText();
+                edit_field_append(console->edit_field, text);
+                return 0;
+            }
+        } break;
```
Comment (rexim) [inline src/ui/console.c]: Could we move the clipboard support implementation to Edit_field so it's available not only in Console? I think it'd be useful.

**Comment by isidentical** [comment]
Thanks for reviews @rexim

**Comment by rexim** [comment]
@isidentical looks good to me! :+1: Thank you for your contribution!

**Pull request merged**